### DISCUSSION TITLE
Populating user's image and name from OAuth provider

### DISCUSSION THREAD

**Discussion by heloineto** [pocketbase/pocketbase#3212] [Show and tell]
Hi there. I'm implementing an authentication flow using Pocketbase and the standard "users" auth collection.

For vanity's sake, I would really like the users that sign in using providers (such as google or github) to have their avatar (profile picture) and name fields filled in when they register.

Here's how I'm doing it currently:

```js
async function authWithProvider(provider) {
  const authData = await pb
    .collection('users')
    .authWithOAuth2({ provider });

  const meta = authData.meta;

  if (meta.isNew) {
    const formData = new FormData();

    const response = await fetch(meta.avatarUrl);

    if (response.ok) {
      const file = await response.blob();
      formData.append('avatar', file);
    }

    formData.append('name', meta.name);

    await pb.collection('users').update(authData.record.id, formData);
  }

  return authData;
}
```

I thought I'd share this code in case someone wants to know how to do this with Pocketbase.
Feel free to tell me if there's something I need to improve in the code.

**Comment by adamshand** [comment]
Very helpful, thanks!

**Comment by vinnycrazzy** [comment]
Thanks!

**Comment by amirrezasalimi** [comment]
thanks

**Comment by shaun49995** [comment]
Any idea why this might fail? It's updating all the details on the record, just not the avatar.

`
const loginWithGoogleOAuth = async () => {
    console.log('Logging In...')

    try {
      setAuthLoading(true)

      const googleAuthData = await pb.collection('users').authWithOAuth2({
        provider: 'google',
        urlCallback: url => {
          console.log('logging the callback')
          WebBrowser.openAuthSessionAsync(url).catch(err => {
            console.log('failed to open url', err)
          })
        },
      })

      console.log('google response: ', googleAuthData)

      const { given_name, family_name } = googleAuthData.meta?.rawUser
      const googleUserName = googleAuthData.meta?.name
      const googleAvatar = googleAuthData.meta?.avatarUrl

      const formData = new FormData()

      if (googleAvatar) {
        const response = await fetch(googleAvatar)
        console.log('google avatar response: ', response)
        if (response.ok) {
          const imageBlob = await response.blob()
          formData.append('avatar', imageBlob)
          console.log('imageBlob: ', imageBlob)
        }
      }

      formData.append('firstName', given_name || googleUserName)
      formData.append('lastName', family_name || '')

      const updatedUserData = googleAuthData.meta?.isNew
        ? await pb
            .collection('users')
            .update(googleAuthData.record.id, formData)
        : googleAuthData.record

      console.log('final form data: ', formData)

      console.log('userData: ', updatedUserData)

      if (updatedUserData) {
        setUser(updatedUserData as UserData)
        await storageUserAndTokenSave(updatedUserData, googleAuthData.token)
      }
    } catch (error) {
      console.error('Failed to log in', error)
      setError('Login failed. Incorrect credentials.')
    } finally {
      setAuthLoading(false)
    }
  }
  `
